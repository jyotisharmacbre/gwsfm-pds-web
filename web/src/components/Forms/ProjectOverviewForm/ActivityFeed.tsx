import React from 'react';
import { FormattedMessage } from 'react-intl';
import { faUser, faCheck } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ProjectApprovalActivityType from '../../../enums/ProjectApprovalActivityType';
import { formatDateAndTime } from '../../../helpers/utility-helper';
import IActivityFeed from '../../../models/IActivityFeed';

const ActivityFeed: React.FC<IActivityFeed> = (props) => {
	return (
		<div className="feed-block" data-test="activity-feed-component">
			{props.activityType == ProjectApprovalActivityType.UserQuery ? (
				<div className="feed-block-img feed-icon" data-test="user-icon">
					<FontAwesomeIcon className="" icon={faUser} />
				</div>
			) : (
					<div className="feed-block-img check-icon" data-test="approved-icon">
						<FontAwesomeIcon className="" icon={faCheck} />
					</div>
				)}
			<div className="feed-block-content">
				<h2>
					<FormattedMessage
						id={
							props.activityType == ProjectApprovalActivityType.UserQuery ? (
								'LABEL_COMMENT_FROM'
							) : (
									'LABEL_APPROVED_BY'
								)
						}
					/>{' '}
					<span>{props.approvedBy}</span>
				</h2>
				<span className="feed-date-time" data-test="formated-date">
					{formatDateAndTime(props.createdDate)}
				</span>
				{/* <p data-test="activity-query">{props.query}</p> */}
				{props.query.split("\n").map((i, key) => {
					return <div data-test="activity-query" key={key}>{i}</div>;
				})
				}
			</div>
		</div>
	);
};

export default ActivityFeed;
