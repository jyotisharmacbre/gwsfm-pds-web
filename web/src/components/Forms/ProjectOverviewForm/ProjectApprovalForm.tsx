import React from 'react';
import { FormattedMessage } from 'react-intl';
import NewTypeAhead from '../../TypeAhead/NewTypeAhead';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle, faClock, faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import { ProjectSignOffStatus } from '../../../store/ProjectOverviewForm/Types/ProjectApprovalEnums';
import ProjectStatus from '../../../enums/ProjectStatus';

interface IProps {
	fields: any;
	formatUserData: any;
	getListOfUsers: (value: any) => Promise<any>;
	status: number
}

const ProjectApprovalForm: React.FC<IProps> = (props) => {
	const { fields, formatUserData, getListOfUsers } = props;
	const getClassAndIcon = (approverType) => {
		let className: string = '', iconType: any;
		let labelID: string = '';
		switch (approverType) {
			case ProjectSignOffStatus.Approved:
				className = "green";
				iconType = faCheckCircle;
				labelID = "LABEL_APPROVED"
				break;
			default:
				className = "orange";
				iconType = faClock;
				labelID = "LABEL_PENDING";
				break;
		}
		return {
			className: className,
			iconType: iconType,
			labelKey: labelID
		};
	}
	const shouldDisplay = (currVal) => {
		return currVal.userId && currVal.projectApprovalId && props.status == ProjectStatus.InReview
	};
	return (
		<div>
			<div className="row">
				<div className="col-md-12 d-flex">
					<label />
					<h6 className="mb-0 d-none">
						<FormattedMessage id="LABEL_SIGN_OFF_STATUS" />{' '}
					</h6>
				</div>
			</div>
			{fields.map((member, index) => (
				<div className="row align-items-stretch" key={index} data-test="project-approval-form">
					<div className="col-11">
						<div className="form-group">
							<NewTypeAhead
								name={`${member}.userId`}
								onSearch={getListOfUsers}
								formatData={formatUserData}
								DynamicsType={fields.get(index).approverTypeDescription}
								labelName={
									fields.get(index).showRangeLabel ? (
										fields.get(index).projectApprovalRangeDescription
									) : null
								}
								submitParam="email"
							/>
							<span className="right_fix_txt">{fields.get(index).approverTypeDescription}</span>
						</div>
					</div>
					<div className="col-1 pl-0">
						{shouldDisplay(fields.get(index)) && <div className="approve_state ">
							<div className="form-group d-flex">
							{<span data-test="icons-approval-status" className="icon"><FontAwesomeIcon className={getClassAndIcon(fields.get(index).approvalStatus).className} icon={getClassAndIcon(fields.get(index).approvalStatus).iconType} /></span>}
							{<label className='approv_label'><FormattedMessage id={getClassAndIcon(fields.get(index).approvalStatus).labelKey} /> </label>}
						</div></div>
						}	</div>
				</div>
			))}
		</div>
	);
};

export default ProjectApprovalForm;
