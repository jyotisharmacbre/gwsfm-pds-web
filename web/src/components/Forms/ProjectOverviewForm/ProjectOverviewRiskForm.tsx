import React from 'react';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import { Field } from 'redux-form';

interface IProps {
    riskName: string;
    riskLabelName: string;
    riskControlMeasureName: string;
    riskControlMeasureLabelName: string;
}

const ProjectOverviewRiskForm: React.FC<IProps> = props => {
    return (
        <div>
            <Field
                name={props.riskName}
                data-test="riskName"
                type="text"
                component={PdsFormInput}
                labelKey={props.riskLabelName}
            />
            <Field
                name={props.riskControlMeasureName}
                data-test="riskControlMeasureName"
                type="text"
                component={PdsFormInput}
                labelKey={props.riskControlMeasureLabelName}
            />
        </div>
    );
};
export default ProjectOverviewRiskForm;
