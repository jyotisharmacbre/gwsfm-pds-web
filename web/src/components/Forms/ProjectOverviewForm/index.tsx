import React from 'react';
import {
  Grid,
  FormControl,
  InputLabel,
  TextField,
  makeStyles,
  createStyles,
  Theme,
  RadioGroup,
  FormControlLabel,
  Radio,
  MenuItem
} from '@material-ui/core';
import { PageBtnActions } from '../../BtnActions/BtnActions';
import { IBtnActionProps } from '../../../props/AppProps';
import SquareRadio from '../../Buttons/SquareRadio';
import { ISquareRadioProps } from '../../Buttons/SquareRadio/props';
import DateContainerForm from '../DateContainerForm';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3, 2)
    },
    form: {
      width: '100%', // Fix IE 11 issue.
      marginTop: theme.spacing(3)
    },
    submit: {
      margin: theme.spacing(3, 0, 2)
    },
    textField: {
      backgroundColor: 'white'
    },
    menu: {
      width: 200
    },
    label: {
      position: 'relative',
      color: '#5b5b5b',
      fontWeight: 'bold'
    },
    select: {
      //marginTop: '0px'
    },
    formControl: {
      margin: theme.spacing(3)
    }
  })
);

const workTypes = [
  { name: 'Select', value: '' },
  { name: 'val 1', value: 'val1' }
];
const projectBudget = [
  { name: 'Select', value: '' },
  { name: 'Up to Â£3 million', value: '3million' }
];
const radio: ISquareRadioProps[] = [
  { value: 'true', label: 'Yes', id: 'cmdyes' },
  { value: 'false', label: 'No', id: 'cmdno' }
];
const radioProject: ISquareRadioProps[] = [
  { value: 'true', label: 'Yes', id: 'plyes' },
  { value: 'false', label: 'No', id: 'plno' }
];

const Buttons: IBtnActionProps[] = [
  {
    Title: 'Previous',
    Color: 'back',
    HandleClick: () => {
      window.location.href = '/';
    }
  },
  {
    Title: 'Next',
    Color: 'cbregreen',
    isSubmit: true,
    HandleClick: () => {
      window.location.href = '/';
    }
  }
];

const ProjectOverviewForm: React.FC = (props: any) => {
  const classes = useStyles();

  const handleValueChange = (name: string) => (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    let val = event.target.value;
    let data = { ...props.form, [name]: val };
    console.log(data);
  };

  return (
    <React.Fragment>
      <Grid spacing={3} container>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Main Contractor
            </InputLabel>

            <TextField
              id="main-contractor"
              inputProps={{
                className: classes.textField
              }}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Enquiry Received Form
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Potential Customer
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl component="fieldset" fullWidth>
            <InputLabel shrink className={classes.label}>
              Types of Enquiry
            </InputLabel>
            <RadioGroup aria-label="gender" name="gender2">
              <FormControlLabel
                value="written"
                control={<Radio color="primary" />}
                label="Written"
              />
              <FormControlLabel
                value="verbal"
                control={<Radio color="primary" />}
                label="Verbal"
              />
              <FormControlLabel
                value="enquiryspecificationorgs"
                control={<Radio color="primary" />}
                label="Full tender enquiry with specifications &amp; orgs"
              />
              <FormControlLabel
                value="other"
                control={<Radio color="primary" />}
                label="Other"
              />
            </RadioGroup>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Project Manager
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Credit Check Result
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl component="fieldset" fullWidth>
            <InputLabel shrink className={classes.label}>
              CMD Notifiable
            </InputLabel>
            <SquareRadio Radios={radio} />
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Form of Contract
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Retention
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Liquidated Damages
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Insurance
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={6}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Work Type
            </InputLabel>
            <TextField
              id="outlined-select-worktype"
              select
              className={classes.textField}
              onChange={handleValueChange('worktype')}
              SelectProps={{
                MenuProps: {
                  className: classes.menu
                }
              }}
              margin="normal"
              name="WorkType"
              variant="outlined"
              required
              fullWidth
              autoFocus
              // value={props.form.customer_contract}
              //error={props.form.invalidCustomerContract}
              //helperText={props.form.invalidCustomerContract ? 'This field is required' : ''}
            >
              {workTypes.map(option => (
                <MenuItem key={option.value} value={option.value}>
                  {option.name}
                </MenuItem>
              ))}
            </TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Project Plan
            </InputLabel>
            <DateContainerForm
              dateOneLabel="Commence Date"
              dateTwoLabel="Completion Date"
            >
              <FormControl fullWidth>
                <InputLabel shrink className={classes.label}>
                  Project Milestone
                </InputLabel>

                <TextField
                  id="main-contractor"
                  multiline
                  rows={4}
                  className={classes.textField}
                  onChange={handleValueChange('maincontractor')}
                  value={''}
                  margin="normal"
                  variant="outlined"
                  required
                  fullWidth
                  autoFocus
                  error={false}
                ></TextField>
              </FormControl>
            </DateContainerForm>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Project Plan
            </InputLabel>
            <DateContainerForm
              dateOneLabel="First Valuation Date"
              dateTwoLabel="Final Account Date"
            >
              <FormControl fullWidth>
                <InputLabel shrink className={classes.label}>
                  Valuation Intervals
                </InputLabel>

                <TextField
                  id="main-contractor"
                  className={classes.textField}
                  onChange={handleValueChange('maincontractor')}
                  value={''}
                  margin="normal"
                  variant="outlined"
                  required
                  fullWidth
                  autoFocus
                  error={false}
                ></TextField>
              </FormControl>
              <FormControl fullWidth>
                <InputLabel shrink className={classes.label}>
                  Payment Terms
                </InputLabel>

                <TextField
                  id="main-contractor"
                  className={classes.textField}
                  onChange={handleValueChange('maincontractor')}
                  value={''}
                  margin="normal"
                  variant="outlined"
                  required
                  fullWidth
                  autoFocus
                  error={false}
                ></TextField>
              </FormControl>
            </DateContainerForm>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Authorised By
            </InputLabel>

            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
            <br />
            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
            <br />
            <TextField
              id="main-contractor"
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={6}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Project Budget
            </InputLabel>
            <TextField
              id="outlined-select-worktype"
              select
              className={classes.textField}
              onChange={handleValueChange('worktype')}
              SelectProps={{
                MenuProps: {
                  className: classes.menu
                }
              }}
              margin="normal"
              name="WorkType"
              variant="outlined"
              required
              fullWidth
              autoFocus
              // value={props.form.customer_contract}
              //error={props.form.invalidCustomerContract}
              //helperText={props.form.invalidCustomerContract ? 'This field is required' : ''}
            >
              {projectBudget.map(option => (
                <MenuItem key={option.value} value={option.value}>
                  {option.name}
                </MenuItem>
              ))}
            </TextField>
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl component="fieldset" fullWidth>
            <InputLabel shrink className={classes.label}>
              Project Is Live
            </InputLabel>
            <SquareRadio Radios={radioProject} />
          </FormControl>
        </Grid>
        <Grid item xs={12} sm={12}>
          <FormControl fullWidth>
            <InputLabel shrink className={classes.label}>
              Comments
            </InputLabel>

            <TextField
              id="main-contractor"
              multiline
              rows={4}
              className={classes.textField}
              onChange={handleValueChange('maincontractor')}
              value={''}
              margin="normal"
              variant="outlined"
              required
              fullWidth
              autoFocus
              error={false}
            ></TextField>
          </FormControl>
        </Grid>

        <Grid justify="space-between" container spacing={3}>
          <Grid item xs={12} sm={12} className="actions">
            <div className="leftalign">
              <PageBtnActions Actions={[Buttons[0]]} />
            </div>
            <div className="rightalign">
              <PageBtnActions Actions={[Buttons[1]]} />
            </div>
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

export default ProjectOverviewForm;
