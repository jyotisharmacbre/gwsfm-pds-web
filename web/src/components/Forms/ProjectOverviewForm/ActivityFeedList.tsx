import React, { useState, useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import ActivityFeed from './ActivityFeed';
import ProjectApprovalActivityType from '../../../enums/ProjectApprovalActivityType';
import { IProjectApprovalActivitiy } from '../../../store/ProjectOverviewForm/Types/IProjectApprovalActivitiy';
import { IProjectApprovals } from '../../../store/ProjectOverviewForm/Types/IProjectApprovals';
import IActivityFeed from '../../../models/IActivityFeed';
import * as actions from '../../../store/rootActions';
import * as services from '../../../services';
import { ILookup } from '../../../store/Lookups/Types/ILookup';
import { ProjectApproverTypeAndRangeMapping } from '../../../store/ProjectOverviewForm/Types/ProjectApproverTypeAndRangeMapping';
import { LookupType } from '../../../store/Lookups/Types/LookupType';
import { IUserServiceData } from '../../../store/UserService/Types/IUserService';
import { displayUserName } from '../../../helpers/utility-helper';
interface IProps {
	currencySymbol: string;
	handleGetUserNamesForEmails: (emails: Array<string>) => void;
}

interface IMapStateToProps {
	projectActivities: Array<IProjectApprovalActivitiy>;
	lookups: Array<ILookup>;
}

const ActivityFeedList: React.FC<IProps & IMapStateToProps> = (props) => {
	const [ activityFeedData, setActivityFeedData ] = useState<Array<IActivityFeed>>([]);
	const [ userNamesForEmails, setUserNamesForEmails ] = useState<Array<IUserServiceData>>([]);
	useEffect(
		() => {
			if (props.projectActivities.length > 0) {
				let emails: Array<string> = [];
				props.projectActivities.map((data, index) => {
					emails.push(data.userId);
				});
				services.getUsersForEmailsService(emails).then((response) => {
					setUserNamesForEmails(response.data);
				});
			}
		},
		[ props.projectActivities ]
	);
	useEffect(
		() => {
			if (userNamesForEmails.length > 0 && props.lookups) {
				if (props.projectActivities.length > 0) {
					let activityFeed: Array<IActivityFeed> = [];
					props.projectActivities.map((data, index) => {
						let username = filterUserByEmailId(userNamesForEmails, data.userId);
						activityFeed.push({
							activityType: data.activityType,
							approvedBy: username,
							query: generateQuery(
								data.query,
								username,
								props.currencySymbol,
								data.activityType,
								data.approverType,
								props.lookups,
								ProjectApproverTypeAndRangeMapping
							),
							createdDate: data.createdOn
						});
						setActivityFeedData(activityFeed);
					});
				}
			}
		},
		[ props.lookups, props.projectActivities, userNamesForEmails ]
	);
	const filterUserByEmailId = (data, emailId) => {
		let userName = '';
		let filter = data.filter((ele) => ele.email.toLowerCase() == emailId.toLowerCase());
		if (filter != undefined && filter[0] != undefined)
			userName = displayUserName(filter[0].firstname, filter[0].lastName);
		return userName;
	};
	const generateQuery = (query, username, currencySymbol, activityType, approverType, lookups, mapping) => {
		if (activityType == ProjectApprovalActivityType.SystemGenerated) {
			let approver = lookups.filter(
				(ele) => ele.lookupItem == LookupType.Project_Approver_Type && ele.lookupKey == approverType
			)[0].description;
			let reQuery = '';
			let template = mapping
				.filter((ele) => ele.type == approverType)[0]
				.approvalActivityRangeDesc.replace('{UserName}', username)
				.replace('{CurrencySymbol}', currencySymbol)
				.replace('{ApproverType}', approver)
				.toString();
			query = template;
		}
		return query;
	};

	return (
		<React.Fragment>
			{activityFeedData.length > 0 && (
				<h3 className="feed_head">
					<FormattedMessage id="LABEL_ACTIVITY_FEED" />
				</h3>
			)}
			<section className="activity_feed">
				{activityFeedData &&
					activityFeedData.map((data) => (
						<ActivityFeed
							activityType={data.activityType}
							approvedBy={data.approvedBy}
							query={data.query}
							createdDate={data.createdDate}
						/>
					))}
			</section>
		</React.Fragment>
	);
};

const mapStateToProps = (state: IState) => ({
	projectActivities: state.projectOverview.projectActivities.data,
	lookups: state.lookup.lookups
});

export default connect(mapStateToProps)(ActivityFeedList);
