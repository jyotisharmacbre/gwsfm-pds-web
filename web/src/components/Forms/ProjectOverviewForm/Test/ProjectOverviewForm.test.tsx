import React from 'react';
import { mount, ShallowWrapper } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../../../../store';
import ProjectOverviewForm from '../ProjectOverviewForm';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import {globalIntl} from '../../../../App';
import App from '../../../../App';
describe('ProjectOverviewForm Fields', () => {
  let wrapper: any;
  const props: any = {
    handleSubmit: jest.fn()
  };
  beforeEach(() => {
    jest.mock('../../../../App');   
    globalIntl.formatMessage = jest.fn().mockImplementation(() => {
      return 'intlmessage';
    });
    wrapper = mount(
      <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>       
          <ProjectOverviewForm {...props} />        
        </IntlProvider>
      </Provider>
    );
  });
  it('Defines the component', () => {
    expect(wrapper).toBeDefined();
  });

  describe('Dfines the Form', () => {
    let form: ShallowWrapper;
    beforeEach(() => {
      form = wrapper.find('[form="projectOverviewForm"]').first();
    });
    it('Renders form component', () => {
      expect(form).toHaveLength(1);
    });
  });
  describe('Defines form fields', () => {
    let field: ShallowWrapper;
    describe('Main Contractor field', () => {
      beforeEach(() => {
        field = wrapper.find('input[name="mainContractor"]').first();
      });
      it('Should renders Main Contractor field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Main Contractor is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Enquiry Received From field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="enquiryReceivedFrom"]').first();
      });
      it('Should renders Enquiry Received From field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Enquiry Received From is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Potential Customer field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="potentialCustomer"]').first();
      });
      it('Should renders Potential Customer field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Potential Customer is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Credit Check Result field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="creditCheckResult"]').first();
      });
      it('Should renders Credit Check Result field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Credit Check Result is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Site Address field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="siteAddress"]').first();
      });
      it('Should renders Site Address field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Site Address is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Form Of Contract field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="formOfContract"]').first();
      });
      it('Should renders Form Of Contract field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Form Of Contract is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Retention field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="retention"]').first();
      });
      it('Should renders Retention field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Retention is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Liquidated Damages field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="liquidatedDamages"]').first();
      });
      it('Should renders Liquidated Damages field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Liquidated Damages is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Insurance field', () => {
      let field: ShallowWrapper;
      beforeEach(() => {
        field = wrapper.find('input[name="insurance"]').first();
      });
      it('Should renders Insurance field', () => {
        expect(field.prop('type')).toBe('text');
      });
      it('Shows error when Insurance is set to blank', () => {
        field.simulate('blur');
        const errorBlock = wrapper.find('.text-danger');
        expect(errorBlock).toHaveLength(1);
      });
    });
    describe('Next button', () => {
      beforeEach(() => {
        field = wrapper.find('button[name="next"]').first();
      });
      it('Should renders next button', () => {
        expect(field.prop('type')).toBe('button');
      });
    });
  });
});
