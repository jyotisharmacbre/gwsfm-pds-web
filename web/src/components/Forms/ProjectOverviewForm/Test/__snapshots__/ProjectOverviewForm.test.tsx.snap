// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectOverviewForm Fields Defines form fields Project form reducer should handle GET_PROJECT_DETAIL_ERROR with and return initialState 1`] = `
Object {
  "error": Object {
    "success": false,
  },
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "date",
    "comments": "",
    "completionDate": "date",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": 1,
    "finalAccountDate": "date",
    "firstValuationDate": "date",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "paymentTerms": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 2,
}
`;

exports[`ProjectOverviewForm Fields Defines form fields Project form reducer should handle add project overview successfully 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "date",
    "comments": "",
    "completionDate": "date",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": 1,
    "finalAccountDate": "date",
    "firstValuationDate": "date",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "paymentTerms": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`ProjectOverviewForm Fields Defines form fields Project form reducer should handle edit project overview successfully 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "date",
    "comments": "",
    "completionDate": "date",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": 1,
    "finalAccountDate": "date",
    "firstValuationDate": "date",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "paymentTerms": "",
    "projectAddDetailId": "",
    "projectId": "",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`ProjectOverviewForm Fields Defines form fields Project form reducer should handle get get additional details successfully 1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "authorizedBy": "",
    "authorizedByHop": "",
    "authorizedBySecond": "",
    "authorizedByThird": "",
    "budget": 1,
    "cdmNotifiable": false,
    "commenceDate": "date",
    "comments": "",
    "completionDate": "date",
    "creditCheckResult": "",
    "enquiryReceivedFrom": "",
    "enquiryTypeId": 1,
    "finalAccountDate": "date",
    "firstValuationDate": "date",
    "formOfContract": "",
    "insurance": "",
    "isProjectLive": false,
    "liquidatedDamages": "",
    "mainContractor": "",
    "milestones": "",
    "paymentTerms": "",
    "projectAddDetailId": "",
    "projectId": "1",
    "retention": "",
    "siteAddress": "",
    "valuationIntervals": "",
    "workTypeId": -1,
  },
  "loading": false,
  "notify": 0,
}
`;
